src/pages/index.js

```diff
@@ -1,27 +1,31 @@
 import Link from "next/link";
 
-export default function TransactionList({ initialData}) {
+export default function TransactionList({ transactions }) {
   // Sort transactions by date in descending order
-  const sortedTransactions = initialData.sort(
-    (a, b) => new Date(b.date) - new Date(a.date)
+  const sortedTransactions = transactions.sort(
+    (a, b) => new Date(b.date) - new Date(a.date),
```

Comment:
go either for .toSorted or also add slice, else you are mutating transactions, which can lead to unexpected behavior.

Reply:
Thanks! Fixed it.